assets/wizards/advertising/views/ad-unit/index.js

```diff
@@ -32,12 +31,31 @@ class AdUnit extends Component {
 	/**
 	 * Handle an update to an ad unit field.
 	 *
-	 * @param {string} key Ad Unit field
-	 * @param {any}  value New value for field
+	 * @param {string|Object} adUnitChangesOrKey Ad Unit field name or object containing changes.
+	 * @param {any}           value              New value for field.
+	 *
 	 */
-	handleOnChange( key, value ) {
+	handleOnChange( adUnitChangesOrKey, value ) {
 		const { adUnit, onChange, service } = this.props;
-		onChange( { ...adUnit, ad_service: service, [ key ]: value } );
+		const adUnitChanges =
+			typeof adUnitChangesOrKey === 'string'
+				? { [ adUnitChangesOrKey ]: value }
+				: adUnitChangesOrKey;
+		onChange( { ...adUnit, ad_service: service, ...adUnitChanges } );
+	}
+
+	getSizeOptions() {
+		const { adUnit } = this.props;
+		const sizes = adUnit.sizes && Array.isArray( adUnit.sizes ) ? adUnit.sizes : [];
+		let sizeOptions = [ ...sizes ];
+		if ( adUnit.fluid ) {
+			sizeOptions = [ ...sizeOptions, 'fluid' ];
```

Comment:
This makes the UI shift when a non-last ad unit size is changed to fluid size (this size jumps to the end of the list. This is minor issue though.

Reply:
We would have to send the index of the changed size and create the options from that. I think it's an extra complexity that is not worth the gained UX.